homeworks/rostyslav.havuka_RostyslavHavuka/6-typeScript/src/index.ts

```diff
@@ -0,0 +1,247 @@
+const scoreElement = document.querySelector('[data-score]') as HTMLHeadingElement;
+const cubeElement = document.querySelector('[data-cube]') as HTMLDivElement;
+const cubeScoreElement = document.querySelector('[data-cube-score]') as HTMLDivElement;
+const keyElement = document.querySelector('[data-key]') as HTMLInputElement;
+const progressBarElement = document.querySelector('[data-progress-bar]') as HTMLDivElement;
+const keyStart = document.querySelector('[data-key-start]') as HTMLButtonElement
+const keyStop = document.querySelector('[data-key-stop]') as HTMLButtonElement
+const keyAgain = document.querySelector('[data-key-again]') as HTMLButtonElement
+
+class Game {
+  private currentKey: string = '';
+  private score: number = 100;
+  private interval: number = 2000;
+  private intervalId: NodeJS.Timeout;
+  private intervalStatusBarId: NodeJS.Timeout;
+  private gameState: boolean = false;
+  private timerProgressBar: number = 0;
+  private width: number;
+
+  constructor(
+    private scoreElement: HTMLHeadingElement,
+    private cubeScoreElement: HTMLDivElement,
+    private keyElement: HTMLInputElement,
+    private cubeElement?: HTMLDivElement,
+    private progressBarElement?: HTMLDivElement,
+  ) { }
+
+  start() {
+    window.addEventListener('load', this.startKeysInterval())
+
+    this.subscribeOnKeyPress()
+
+    keyStart.addEventListener('click', (): any => {
+      if (this.gameState === false) {
+        this.width = 1;
+        this.progressBar();
+        this.progressBarElement.classList.remove('hidden')
+        clearInterval(this.intervalId)
+        this.intervalId = setInterval(() => {
+          this.subScoreTimer(this.score)
+          this.sizeDown()
+          this.width = 1;
+          this.progressBar();
+        }, 3000);
+      }
+      this.gameState = true
+    })
+
+    keyStop.addEventListener('click', (): any => {
+      this.gameState = false;
+      this.progressBarElement.classList.add('hidden')
+      clearInterval(this.intervalStatusBarId)
+      this.timerProgressBar = 0;
+      this.width = 99;
+      clearInterval(this.intervalId)
+      alert('your Score:' + this.score)
+      this.score = 100;
+      cubeScoreElement.innerText = this.score.toString()
+      this.scoreElement.innerText = '';
+    })
+
+    keyAgain.addEventListener('click', (): any => {
+      if (this.gameState === true) {
+        this.score = 100;
+        this.width = 1;
+        this.progressBar();
+        cubeScoreElement.innerText = this.score.toString()
+        this.scoreElement.innerText = '';
```

Comment:
Same here. `this.scoreElement.innerText = ...` repeats 4 times! Create a separe `resetScore` method